knack/config.py

```diff
@@ -141,14 +141,21 @@ def set_value(self, section, option, value):
     def set_to_use_local_config(self, use_local_config):
         self.use_local_config = use_local_config
 
+    def remove_option(self, section, option):
+        for config in self._config_file_chain if self.use_local_config else self._config_file_chain[-1:]:
+            if config.remove_option(section, option):
+                return True
+        return False
+
 
 class _ConfigFile(object):
     _BOOLEAN_STATES = {'1': True, 'yes': True, 'true': True, 'on': True,
                        '0': False, 'no': False, 'false': False, 'off': False}
 
-    def __init__(self, config_dir, config_path):
+    def __init__(self, config_dir, config_path, config_notice=None):
         self.config_dir = config_dir
         self.config_path = config_path
+        self.config_notice = config_notice
```

Comment:
Let's call it `config_comment`, and also add some comments about its functionality, including some examples, to make sure `;` is added correctly. 

```suggestion
        self.config_comment = config_comment
```